src/writer/update-proposal.ts

```diff
@@ -0,0 +1,115 @@
+import isEqual from 'lodash/isEqual';
+import snapshot from '@snapshot-labs/snapshot.js';
+import { jsonParse } from '../helpers/utils';
+import db from '../helpers/mysql';
+import { getSpace, getProposal } from '../helpers/actions';
+import log from '../helpers/log';
+import {
+  flaggedAddresses,
+  flaggedProposalTitleKeywords,
+  flaggedProposalBodyKeywords
+} from '../helpers/moderation';
+
+export function validateChoices({ type, choices }): boolean {
+  if (type && choices.length > 0) {
+    switch (type) {
+      case 'basic':
+        return isEqual(['For', 'Against', 'Abstain'], choices);
+      default:
+        return choices.length > 0;
+    }
+  } else {
+    return false;
+  }
+}
+
+// We don't need most of the checks used https://github.com/snapshot-labs/snapshot-sequencer/blob/89992b49c96fedbbbe33b42041c9cbe5a82449dd/src/writer/proposal.ts#L62
+// because we assume that those checks were already done during the proposal creation
+export function getSpaceUpdateError({ type, space }): string | undefined {
+  const { voting = {} } = space;
+
+  if (voting.type && type !== voting.type) return 'space voting type mismatch';
+
+  return undefined;
+}
+
+export function isScamDetected({ address, name, body }): boolean {
+  const proposalNameLC = name.toLowerCase();
+  const proposalBodyLC = body.toLowerCase();
+  const addressLC = address.toLowerCase();
+
+  return (
+    flaggedAddresses.includes(addressLC) ||
+    flaggedProposalBodyKeywords.some(keyword => proposalBodyLC.includes(keyword)) ||
+    flaggedProposalTitleKeywords.some(keyword => proposalNameLC.includes(keyword))
+  );
+}
+
+export async function verify(body): Promise<any> {
+  const msg = jsonParse(body.msg);
+  const timestampNow = Math.floor(Date.now() / 1e3);
+
+  const schemaIsValid: any = snapshot.utils.validateSchema(
+    snapshot.schemas.updateProposal,
```

Comment:
<img width="704" alt="Untitled 2" src="https://github.com/snapshot-labs/snapshot-sequencer/assets/15967809/9b9d894c-1d54-455c-9839-74efc6efc8c9">

Reply:
yes, it's because we need to merge https://github.com/snapshot-labs/snapshot.js/pull/895 first